processor/src/main/resources/org/mapstruct/ap/internal/model/BeanMappingMethod.ftl

```diff
@@ -76,9 +76,12 @@
     	</#if>
     	<@includeModel object=callback targetBeanName=resultName targetType=resultType/>
     </#list>
-    <#if returnType.name != "void">
```

Comment:
So, currently it's this:
```
if( returnType.isMappedByBuilder() ) {
...
}
else if( returnType.name != "void" ) {
...
}
```
and are you saying it should be this?

```
if( returnType.name != "void" ) {
  if ( returnType.isMappedByBuilder() ) {

  }
  else {
    
  }
}
```

I think that makes sense - you can read my updated initial comment - at the bottom I talked about ```@AfterMapping``` methods, which might use a builder and return void.